unidist/core/backends/mpi/core/executor.py

```diff
@@ -563,6 +563,58 @@ def shutdown():
         e_logger.debug("Shutdown rank {}".format(rank_id))
 
 
+def cluster_resources():
+    """
+    Get resources of MPI cluster.
+
+    Returns
+    -------
+    dict
+        Dictionary with cluster nodes info in the form '{node_ip0: {CPU: x0},
+        node_ip1: {CPU: x1}, ..}'.
+    """
+    garbage_collector.regular_cleanup()
+
+    def get_worker_resources():
+        import socket
+
+        hostname = socket.gethostname()
+        host = socket.gethostbyname(hostname)
+        return host
+
+    output_ids = [
+        object_store.generate_output_data_id(
+            dest_rank, garbage_collector, num_returns=1
+        )
+        for dest_rank in range(2, world_size)
+    ]
+
+    operation_type = common.Operation.EXECUTE
+    operations_data = [
+        {
+            "task": get_worker_resources,
+            "args": [],
+            "kwargs": {},
+            "output": common.master_data_ids_to_base(output_ids[idx]),
+        }
+        for idx in range(len(output_ids))
+    ]
+
+    for idx, dest_rank in enumerate(range(2, world_size)):
+        communication.send_remote_task_operation(
+            comm, operation_type, operations_data[idx], dest_rank
+        )
+        garbage_collector.increment_task_counter()
+
+    hosts = get(output_ids)
+
+    cluster_resources = defaultdict(lambda: {"CPU": 0})
+    for host in hosts:
+        cluster_resources[host]["CPU"] += 1
+
+    return dict(cluster_resources)
```

Comment:
Is is a conversion from `defaultdict` to `dict`? Want to clarify, is it necessary?

Reply:
It's necessary to have habitual dictionary view and be aligned with style of all backends:

```python
In [2]: from collections import defaultdict

In [3]: d = defaultdict(lambda: -5)

In [4]: d["a"]+=3

In [5]: d
Out[5]: defaultdict(<function __main__.<lambda>()>, {'a': -2})

In [6]: d_dict = dict(d)

In [7]: d_dict
Out[7]: {'a': -2}
```

Reply:
Thanks for explanation!